src/code/z_vimode.c

```diff
@@ -1,23 +1,346 @@
+    viMode->fldRegs[1].origin = ORIGIN(width * 2 * (fb16Bit ? 1 : 2) * (loRes ? 1 : 2));
+
+    viMode->fldRegs[0].yScale = yScaleLo | yScaleHiEvenField;
+    viMode->fldRegs[1].yScale = yScaleLo | yScaleHiOddField;
+
+    viMode->fldRegs[0].vIntr = 2;
+    viMode->fldRegs[1].vIntr = 2;
+}
+
+void ViMode_Save(ViMode* viMode) {
+    R_VI_MODE_EDIT_STATE = viMode->editState;
+    R_VI_MODE_EDIT_WIDTH = viMode->viWidth;
+    R_VI_MODE_EDIT_HEIGHT = viMode->viHeight;
+    R_VI_MODE_EDIT_ULY_ADJ = viMode->upperAdjust;
+    R_VI_MODE_EDIT_LRY_ADJ = viMode->lowerAdjust;
+    R_VI_MODE_EDIT_ULX_ADJ = viMode->leftAdjust;
+    R_VI_MODE_EDIT_LRX_ADJ = viMode->rightAdjust;
+
+    if (SREG(58) == 1) {
+        SREG(58) = 0;
+        switch (SREG(59)) {
+            case 1:
+                ViMode_LogPrint(&osViModePalLan1);
+                break;
+            case 2:
+                ViMode_LogPrint(&osViModeFpalLan1);
+                break;
+            default:
+                ViMode_LogPrint(&viMode->customViMode);
+                break;
+        }
+    }
+}
+
+void ViMode_Load(ViMode* viMode) {
+    if ((R_VI_MODE_EDIT_WIDTH & ~3) == 1) {
+        R_VI_MODE_EDIT_WIDTH += 4;
+    }
+
+    viMode->editState = R_VI_MODE_EDIT_STATE;
+    viMode->viWidth = R_VI_MODE_EDIT_WIDTH & ~3;
+    viMode->viHeight = R_VI_MODE_EDIT_HEIGHT;
+    viMode->upperAdjust = R_VI_MODE_EDIT_ULY_ADJ;
+    viMode->lowerAdjust = R_VI_MODE_EDIT_LRY_ADJ;
+    viMode->leftAdjust = R_VI_MODE_EDIT_ULX_ADJ;
+    viMode->rightAdjust = R_VI_MODE_EDIT_LRX_ADJ;
+}
+
+void ViMode_Init(ViMode* viMode) {
+    viMode->editState = VI_MODE_EDIT_STATE_INACTIVE;
+    viMode->viWidth = SCREEN_WIDTH;
+    viMode->viHeight = SCREEN_HEIGHT;
+    viMode->leftAdjust = 0;
+    viMode->rightAdjust = 0;
+    viMode->upperAdjust = 0;
+    viMode->lowerAdjust = 0;
+    viMode->viFeatures = OS_VI_DITHER_FILTER_ON | OS_VI_GAMMA_OFF;
+    viMode->tvType = osTvType;
+    viMode->fb16Bit = true;
+    viMode->modeN = true;
+    viMode->antialiasOff = false;
+    viMode->loRes = true;
+
+    ViMode_Save(viMode);
+}
+
+void ViMode_Destroy(ViMode* viMode) {
+}
+
+void ViMode_ConfigureFeatures(ViMode* viMode, s32 viFeatures) {
+    u32 ctrl = viMode->customViMode.comRegs.ctrl;
+
+    if (viFeatures & OS_VI_GAMMA_ON) {
+        ctrl |= OS_VI_GAMMA;
+    }
+    if (viFeatures & OS_VI_GAMMA_OFF) {
+        ctrl &= ~OS_VI_GAMMA;
+    }
+    if (viFeatures & OS_VI_GAMMA_DITHER_ON) {
+        ctrl |= OS_VI_GAMMA_DITHER;
+    }
+    if (viFeatures & OS_VI_GAMMA_DITHER_OFF) {
+        ctrl &= ~OS_VI_GAMMA_DITHER;
+    }
+    if (viFeatures & OS_VI_DIVOT_ON) {
+        ctrl |= OS_VI_DIVOT;
+    }
+    if (viFeatures & OS_VI_DIVOT_OFF) {
+        ctrl &= ~OS_VI_DIVOT;
+    }
+    viMode->customViMode.comRegs.ctrl = ctrl;
+}
+
+// This function uses controller input (C buttons + D pad) to reconfigure the custom VI mode
+void ViMode_Update(ViMode* viMode, Input* input) {
```

Comment:
```suggestion
/** 
 * This function uses controller input (C buttons + D pad) to reconfigure the custom VI mode
 */
void ViMode_Update(ViMode* viMode, Input* input) {
```
This is the style we use for function descriptions. Applies to a few places.

Also, I think OoT comments are good, noticed some are removed when the functions seems identical